packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1773,6 +1659,62 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         return summarizeResult as ISummaryTreeWithStats;
     }
 
+    /**
+     * IGarbageCollectionRuntime::getGCData implementation. Returns the GC data for this container.
+     * @param fullGC - true to bypass optimizations and force full generation of GC data.
+     */
+    public async getGCData(fullGC?: boolean): Promise<IGarbageCollectionData> {
+        return this.dataStores.getGCData(fullGC);
+    }
+
+    /**
+     * IGarbageCollectionRuntime::updateUsedRoutes implementation. After GC has run, called to notify this
+     * container's nodes of routes that are used in it.
+     * @param usedRoutes - The routes that are used in all nodes in this Container.
+     * @returns the statistics of the used state of the data stores.
+     */
+    public updateUsedRoutes(usedRoutes: string[]): IUsedStateStats {
+        // Update our summarizer node's used routes. Updating used routes in summarizer node before
+        // summarizing is required and asserted by the the summarizer node. We are the root and are
+        // always referenced, so the used routes is only self-route (empty string).
+        this.summarizerNode.updateUsedRoutes([""]);
+
+        return this.dataStores.updateUsedRoutes(
+            usedRoutes,
+            // For now, we use the timestamp of the last op for gcTimestamp. However, there can be cases where
+            // we don't have an op (on demand summaries for instance). In those cases, we will use the timestamp
+            // of this client's connection - https://github.com/microsoft/FluidFramework/issues/7152.
+            this.deltaManager.lastMessage?.timestamp,
+        );
+    }
+
+    /**
+     * IGarbageCollectionRuntime::deleteUnusedRoutes implementation. Currently only called when this container is
+     * running in GC test mode. It is called to delete objects whose routes are unused. This enables testing scenarios
+     * where the container hase deleted content.
+     * @param unusedRoutes - The routes that are unused in all data stores in this Container.
+     */
+    public deleteUnusedRoutes(unusedRoutes: string[]): void {
+        this.dataStores.deleteUnusedRoutes(unusedRoutes);
```

Comment:
I do not know if it's good idea, but you can potentially reduce scope of IGarbageCollectionRuntime by asking to provide deleteUnusedRoutesCallback in GarbageCollector constructor.
This might be useful if most of such callbacks simply delegate to other objects (i.e. one-liners like this one).

Reply:
I can see why deleteUnusedRoutes shouldn't really be part of the provider contract because its temporary and specific to this impementation.